liblangutil/Scanner.cpp

```diff
@@ -271,12 +274,58 @@ bool Scanner::skipWhitespaceExceptUnicodeLinebreak()
 	return sourcePos() != startPosition;
 }
 
+
+namespace {
+	/// Tries to scan for an RLO/LRO/RLE/LRE/PDF and keeps track of script writing direction override depth.
+	///
+	/// @returns ScannerError::NoError in case of successful parsing and directional encodings are paired
+	///          and error code in case the input's lexical parser state is invalid and this error should be reported
+	///          to the user.
+	static ScannerError isUnicodeSubsequenceWellFormed(CharStream& _stream, size_t _startPosition)
+	{
+		static array<pair<string_view, int>, 5> constexpr m_sequences{
+				pair<string_view, int>{"\xE2\x80\xAD", 1}, // U+202D (LRO - Left-to-Right Override)
+				pair<string_view, int>{"\xE2\x80\xAE", 1}, // U+202E (RLO - Right-to-Left Override)
+				pair<string_view, int>{"\xE2\x80\xAA", 1}, // U+202A (LRE - Left-to-Right Embedding)
+				pair<string_view, int>{"\xE2\x80\xAB", 1}, // U+202B (RLE - Right-to-Left Embedding)
+				pair<string_view, int>{"\xE2\x80\xAC", -1} // PDO - Pop Directional Override
```

Comment:
There's no PDO. The pop that works with both LRO/RLO and LRE/RLE is PDF.
```suggestion
				pair<string_view, int>{"\xE2\x80\xAC", -1} // PDF - Pop Directional Formatting
```